test/passing/break_before_func.ml

```diff
@@ -0,0 +1,195 @@
+[@@@ocamlformat "break-before-func=false"]
+
+;;
+List.fooo (fun x ->
+    xxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxxxxxxxxxxx
+      xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxx
+  )
+
+;;
+List.fooo (fun x ->
+    xxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxxxxxxxxxxx
+      xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxx
+  )
+
+(* foooooooooooo *)
+
+;;
+List.fooo (function
+  | A ->
+      xxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxxxxxxxxxxx
+        xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxx
+  )
+
+;;
+List.fooo (function
+  | A ->
+      xxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxxxxxxxxxxx
+        xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxx
+  )
+
+(* foooooooooo *)
+
+;;
+List.map (fun x ->
+    xxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxxxxxxxxxxx
+      xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxx
+  ) list
+
+;;
+List.map (* toto *)
+  (fun x ->
+    xxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxxxxxxxxxxx
+      xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxx
+  ) list
+
+;;
+List.map (* toto *)
+  (fun x ->
+    xxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxxxxxxxxxxx
+      xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxx
+  ) (* foooooooooo *) list
+
+;;
+List.map
+  (* this is a loooooooooooooooooooooooooooooong commmmmmmmmmment
+     blablablabla *)
+  (fun xxxxxxxxxxxxxxxxxxxxx yyyyyyyyyyyyyyyy zzzzzzzzzzzzzzzzzz
+         aaaaaaaaaaaaaaaa ->
+    xxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxxxxxxxxxxx
+      xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxx
+  ) list
+
+;;
+List.map
+  (* this is a loooooooooooooooooooooooooooooong commmmmmmmmmment
+     blablablabla *)
+  (fun xxxxxxxxxxxxxxxxxxxxx yyyyyyyyyyyyyyyy zzzzzzzzzzzzzzzzzz
+         aaaaaaaaaaaaaaaa ->
+    xxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxxxxxxxxxxx
+      xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxx )
+  (fun xxxxxxxxxxxxxxxxxxxxx yyyyyyyyyyyyyyyy zzzzzzzzzzzzzzzzzz
+       aaaaaaaaaaaaaaaa ->
+    xxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxxxxxxxxxxx
+      xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxx )
+  (fun xxxxxxxxxxxxxxxxxxxxx yyyyyyyyyyyyyyyy zzzzzzzzzzzzzzzzzz
+       aaaaaaaaaaaaaaaa ->
+    xxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxxxxxxxxxxx
+      xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxx ) list
```

Comment:
Not breaking before `list` here is a bit dangerous

Reply:
To expand on this point, elsewhere a lot of effort has been put into avoiding cases where something short can appear on the same line after something that spans multiple lines. Experience shows that it is very easy to visually overlook such code, and by default I think we should keep with that approach and break here too.

Reply:
Should be fixed now